client-s3/src/main/kotlin/app/cash/backfila/client/s3/record/StringStrategies.kt

```diff
@@ -0,0 +1,24 @@
+package app.cash.backfila.client.s3.record
+
+import okio.BufferedSource
+import okio.ByteString
+
+// This file contains some general strategies for handling String files for Backfills.
+
+/**
+ * Produces single string records separated by newlines.
+ */
+class Utf8StringNewlineStrategy : RecordStrategy<String> {
+  override fun calculateNextRecordBytes(source: BufferedSource): Long = newlineByteCalculator(source)
+
+  override fun bytesToRecords(source: ByteString): List<String> = source.utf8().split("\n")
+}
+
+/**
+ * Optimized newline record byte calculator.
+ */
+fun newlineByteCalculator(source: BufferedSource): Long {
```

Comment:
should this be private?

Reply:
maybe.. I don't love that its global scope though

Reply:
Sure I can do that StringStrategyTools?